client/src/components/RoomDirectoryPanel.tsx

```diff
@@ -0,0 +1,184 @@
+            search.addIndex('name')
+            search.addDocuments(AllRooms as IChatRoom[]);
+            let results = search.search(searchText);
+            setDisplayedRooms(results as IChatRoom[]);
+        }
+    };
+
+    // const RoomIcon = ({ room } : IRoomIconProps ) => {
+    //     const [isHovering, setIsHovering] = useState(false);
+    //     const normalIcon = () => <p>{room}</p>;
+    //     const hoveringIcon = () => (
+    //         <div className="display-hover-icon">
+    //             <span>{room}</span>
+    //         </div>
+    //     );
+
+    //     const iconToDisplay = isHovering ? normalIcon : hoveringIcon;
+
+    //     return (
+    //         <div className="room-icon"
+    //             onMouseEnter={() => setIsHovering(true)}
+    //             onMouseLeave={() => setIsHovering(false)}
+    //             onTouchStart={() => setIsHovering(true)}
+    //         >
+    //             <button style={{ background: "none", border: "none" }} onClick={() => sendRoomDirectory('roomDirectory', room)} >
+    //                 {iconToDisplay()}
+    //             </button>
+    //         </div>
+    //     );
+    // }
+
+    const displayRooms = displayedRooms?.map((room, index) => {
+        return (
+            <div key={index} className="room-icon"
+            >
+                <button style={{ background: "none", border: "none" }} onClick={() => sendRoomDirectory('roomDirectory', room.name)} >
+                    <p>{room.name}</p>
+                </button>
+            </div>                        
+        );
+        });
+    
+    return (
+        <div className="room-container">
+            <div style={{ justifyContent: "center", display: 'flex' }} >
+            <InputBase
+                placeholder="Search Rooms"
+                onChange={(event) => setSearchText(event.target.value)}
+                onKeyPress={(event) => event.key === 'Enter' && onSearchSubmit()}
+                value={searchText}
+                />
+            <IconButton color="primary" onClick={onSearchSubmit}>
+                <SearchIcon />
+            </IconButton>
+            <IconButton color="primary" href={'http://www.trychats.com/#/'}>
+                <HomeIcon />
+            </IconButton>
+            </div>
+            <div className="room-icon-list"> {displayRooms} </div>
+        </div>
+    );
+}
+
+export const EnterRoomModal = ({roomName, onClickCancel} : IEnterRoomProps ) => {
+    const classes = useStyles();
+    const buttonClasses = buttonStyles();
+
+    return (
+        <div className={classes.container}>
+            <IconButton onClick={onClickCancel} className={classes.cancelButton}>
+                <Cancel />
+            </IconButton>
+        <div className={classes.title}> {`Enter room: ${roomName}`}</div>
+        <Button
+        variant="contained"
+        color="primary"
+        className={buttonClasses.button}
+        href={`http://www.trychats.com/#/room/${roomName}`}
```

Comment:
instead of a link, you should use `history`
we use `react-router` / `react-router-dom` to implement routes like /rooms/

they allow us to change the user's route but also maintain a history, so we can go back and forth between pages and not have to completely reload the site

when a user clicks a regular link to our site they will reload and also reconnect with the server which we don't really want, so we can use `history` instead

there's an example in App.tsx, 

since all our components are under the react router context, in any component you can do
```
const history = useHistory();
// somewhere in the code like on button click
history.push("/rooms/someRoom")
```